packages/expo-screen-orientation/ios/ScreenOrientationModule.swift

```diff
@@ -0,0 +1,104 @@
+import ExpoModulesCore
+
+public class ScreenOrientationModule: Module, OrientationListener {
+  static let didUpdateDimensionsEvent = "expoDidUpdateDimensions"
+
+  let screenOrientationRegistry = ScreenOrientationRegistry.shared
+  var eventEmitter: EXEventEmitterService?
+
+  public func definition() -> ModuleDefinition {
+    Name("ExpoScreenOrientation")
+
+    Events("expoDidUpdateDimensions")
+
+    AsyncFunction("lockAsync") { (orientationLock: ModuleOrientationLock) in
+      let orientationMask = orientationLock.toInterfaceOrientationMask()
+      guard !orientationMask.isEmpty else {
+        throw InvalidOrientationLockException()
+      }
+
+      if !orientationMask.isSupportedByDevice() {
+        throw UnsupportedOrientationLockException("\(orientationLock.rawValue)")
+      }
+      screenOrientationRegistry.setMask(orientationMask, forModule: self)
+    }
+
+    AsyncFunction("lockPlatformAsync") { (allowedOrientations: [ModuleOrientation]) in
+      var allowedOrientationsMask: UIInterfaceOrientationMask = []
+      for allowedOrientation in allowedOrientations {
+        let orientation = allowedOrientation.toInterfaceOrientation()
+        let orientationMask = orientation.toInterfaceOrientationMask()
+        if orientationMask.isEmpty {
+          throw InvalidOrientationLockException()
+          return
+        }
+
+        allowedOrientationsMask.insert(orientationMask)
+      }
+
+      if !allowedOrientationsMask.isSupportedByDevice() {
```

Comment:
In most cases when you want to exit early from the function when certain conditions are not met, it's better to use `guard..else`
```suggestion
      guard allowedOrientationsMask.isSupportedByDevice() else {
```